src/test/scala/common/KafkaIntSpec.scala

```diff
@@ -21,10 +24,42 @@ abstract class KafkaIntSpec extends BaseSpec with BeforeAndAfterAll with Patienc
   lazy val zkUtils = ZkUtils(s"localhost:${kafkaServer.zookeeperPort}", zkSessionTimeout.toMillis.toInt,
     zkConnectionTimeout.toMillis.toInt, isZkSecurityEnabled = false)
 
+  lazy val kafkaAdminClient = AdminClient.create(Map[String, AnyRef](
+    BOOTSTRAP_SERVERS_CONFIG -> s"localhost:$kafkaPort"
+  ).asJava)
+
+  val defaultTopicProperties = Map(
```

Comment:
why do we need this now?

Reply:
with the new client `KafkaAdminClient` when the topic configurations are fetched, it come with all the defaults... so a topic created without any configs, would come back with the default configs.... in order for assertions to work, a base topic needs to include all the defaults.

Reply:
I have been thinking further about this concern, and the fact is that we can't really construct reliably a `Topic` object out of the configuration file that would match a fetched `Topic` object, so as this stands, the configuration will always be applied...

We could work out something, so rather than a comparing of topics, we compare whether the properties in the configuration are satisfied by the fetched properties...

Thoughts?

Reply:
Ok, Basically consider the following topic configuration
```yaml
topic1:
  partitions: 10
  replication: 3
  config:
    cleanup.policy: compact
    delete.retention.ms: 86400000
    min.compaction.lag.ms: 21600000
    retention.ms: 0
    min.insync.replicas: 3
```

This will get de-serialised into...
```scala
Topic(10, 3, Map(cleanup.policy -> compact, delete.retention.ms -> 86400000, min.compaction.lag.ms -> 21600000, retention.ms -> 0, min.insync.replicas -> 3))
```
...however this is not correct, as the topic will be created with many more properties, all the defaults + the the ones specified.

This really spands into two issues:
-  A topic de-serialised from config, can't be compared for equality with a topic de-serialised from a fetch. So a more sophisticated comparison is needed (i.e. are all the entries in config present in the fetched topic?)
-  The config can't ever be the source of truth, as it doesn't have all the facts. This also raises a question about removing a config entry.